components/apimgt/org.wso2.carbon.apimgt.common.analytics/src/main/java/org/wso2/carbon/apimgt/common/analytics/publishers/impl/AnalyticsDataPublisher.java

```diff
@@ -101,4 +165,20 @@ public CounterMetric getFaultyMetricReporter() {
         }
         return faultyMetricReporter;
     }
+
+    public List<CounterMetric> getSuccessMetricReporters() {
+
+        if (this.successMetricReporters.isEmpty()) {
+            throw new RuntimeException("None of AnalyticsDataPublishers are initialized.");
```

Comment:
Throw a checked exception and handle it propoerly from the caller

Reply:
Ack.